### DISCUSSION TITLE
20220713(수) 백엔드 출시 전, 세부 일정사항

### DISCUSSION THREAD

**Discussion by jungmyeong96** [innovationacademy-kr/42dashboard#197] [Backend Progress]
# 20220713(수)

## 남은 기간

- 다음주 : 아래 할 일 요약
- 다다음주 : 아래 할 일 요약
- 마지막주: 리펙토링 및 에러 및 가이드작성

## 목표

- **데이터가 문제없이 들어오고, 필터링도 문제없이 요청응답되는 상황**
- 입력은 스프레드로 받고, 수정 및 삭제는 admin을 통해 진행예정
    - 스프레드는 컬럼추가나 테이블 추가 시, 생길 수 있는 오류가 있으므로 가이드를 명확히 줄 것
    - db관리 툴 작성요령을 명확히 전달할 것
- aws 백업데이터를 날마다 저장 (덮어쓰기인지, 새로 생성하여 누적인지 확인)
- 데이터를 원하는 대로 추출가능 (엑셀)

### huchoi

- aws에 올릴 때, 도커 컴포즈로 감싸야하는지? 의문
- aws 백업방식 알아보기 및 가이드 작성
- aws에서 데이터 추출하는 방식 알아보기 및 기능 구현
- pgadmin 알아보기
- 프론트에서 요청하는 프리셋 저장
- db관리 툴 사용 가이드만들기
- 테스터 만들기()

### junghan

- 테이블 간의 관계설정 재정립 (ex 지급금액, 총지급)
- 새로 받은 스프레드 데이터 목업데이터만들기 및 파싱규칙 재정비
- 파싱된 데이터 유효성 확인
- 테스터 만들기()
- 외래 키 제약조건 해제 변경
- 외래키 설정 시 직접 파싱후, 데이터 넣어주는 방식 수정요청
- 스프레드 입력 가이드만들기

### doyun

- 일대다 테이블의 경우, ‘다’ 테이블은 validate_date, expired_date 추가 후, 로그데이터의 경우 데이터삽입
- 계산가능한 데이터는 업데이터 측에서 계산하여 컬럼 구성하는 것이 사용성 측면에서 좋다고 생각
    - ex)블랙홀 잔여일자
- 파싱된 데이터 유효성 확인
- 테스터 만들기()

## 현재까지 구현된 코드를 요약한 다이어그램
![image](https://user-images.githubusercontent.com/55140432/178677704-9d86a6a7-5647-4ce5-83c0-1623564abd99.png)

**Comment by hunjin-choi** [comment]
감사합니다 중한님!! 다이어그램 수정하신거 정말 잘 하신거 같습니다!
